src/components/links/Links.tsx

```diff
@@ -54,6 +59,57 @@ function Links() {
 	const boxUpdater = useBoxUpdater();
 
 	const [incoming, outgoing] = boxUpdater.selectedBoxConnections;
+	const [showEditor, setShowEditor] = useState(false);
+	const [editableLink, setEditableLink] = useState<Link<ExtendedConnectionOwner>>();
+
+	const createNewLink = () => {
+		setEditableLink(undefined);
+		setShowEditor(true);
+	};
+
+	const onSubmit = (value: Link<ExtendedConnectionOwner>) => {
+		if (editableLink) {
+			boxUpdater.changeLink(editableLink, value);
+		} else {
+			boxUpdater.addLink(value);
+		}
+	};
+
+	const onDelete = () => {
+		if (editableLink) {
+			boxUpdater.deleteLink(editableLink).then();
+		}
+	};
+
+	const onClose = () => setShowEditor(false);
+
+	const editLink = (direction: 'to' | 'from', box?: BoxEntity, pin?: Pin) => {
```

Comment:
1. Incorrect name. This function opens link editor with pre-filled fields, not edits links.
2. You use this function to open editor when you wan't to edit link, and when you want to add new link by clicking the `New link +` button.
In summary, you need a function like `openLinkEditor` that will replace your `editLink` and `createNewLink` functions